src/compas_view2/objects/bufferobject.py

```diff
@@ -0,0 +1,105 @@
+from compas.utilities import flatten
+
+from ..buffers import make_index_buffer, make_vertex_buffer
+
+from .object import Object
+
+
+class BufferObject(Object):
+    """Object for displaying COMPAS mesh data structures.
+    """
+
+    default_color_points = [0.2, 0.2, 0.2]
+    default_color_lines = [0.4, 0.4, 0.4]
+    default_color_frontfaces = [0.8, 0.8, 0.8]
+    default_color_backfaces = [0.8, 0.8, 0.8]
+
+    def __init__(self, data, name=None, is_selected=False, show_points=False,
+                 show_lines=False, show_faces=False):
+        super().__init__(data, name=name, is_selected=is_selected)
+        self._data = data
+        self._points_buffer = None
+        self._lines_buffer = None
+        self._frontfaces_buffer = None
+        self._backfaces_buffer = None
+        self.show_points = show_points
+        self.show_lines = show_lines
+        self.show_faces = show_faces
+        self.linewidth = 1
+        self.pointsize = 10
+
+    def make_buffers(self):
+
+        if hasattr(self, '_points'):
+            self._points_buffer = {
+                'positions': make_vertex_buffer(list(flatten(self._points))),
+                'colors': make_vertex_buffer(list(flatten(self._pointcolors))),
+                'elements': make_index_buffer([i for i in range(len(self._pointelements))]),
+                'n': len(self._points)
+            }
+
+        if hasattr(self, '_linevertices'):
+            self._lines_buffer = {
+                'positions': make_vertex_buffer(list(flatten(self._linevertices))),
+                'colors': make_vertex_buffer(list(flatten(self._linevertexcolors))),
+                'elements': make_index_buffer(list(flatten(self._lineelements))),
```

Comment:
what happens if these are not defined?

Reply:
so if the buffer does not exist it will skipped to draw. adding this to draw function now...